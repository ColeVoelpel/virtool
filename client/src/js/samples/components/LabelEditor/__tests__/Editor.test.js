import { mapDispatchToProps, mapStateToProps } from "../LabelEditor";
import { EditLabel } from "../EditLabel";
import { CreateLabel } from "../CreateLabel";
import { ColorSelector } from "../ColorSelector";

describe("<EditLabel>", () => {
    it("should render", () => {
        const wrapper = shallow(<EditLabel />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<CreateLabel>", () => {
    it("should render", () => {
        const wrapper = shallow(<CreateLabel />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("<ColorSelector>", () => {
    it("should render", () => {
        const wrapper = shallow(<ColorSelector />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("mapDispatchToProps()", () => {
    const dispatch = jest.fn();
    const props = mapDispatchToProps(dispatch);

    it("should return onLoadLabels() in props", () => {
        props.onLoadLabels();
        expect(dispatch).toHaveBeenCalledWith({ type: "GET_LABELS_REQUESTED" });
    });

    it("should return submitNewLabel() in props", () => {
        props.submitNewLabel("Label 2", "BarFoo", "#FFFFFF");
        expect(dispatch).toHaveBeenCalledWith({
            type: "CREATE_LABEL_REQUESTED",
            name: "Label 2",
            description: "BarFoo",
            color: "#FFFFFF"
        });
    });

    it("should return removeLabel() in props", () => {
        props.removeLabel("UniqueID");
        expect(dispatch).toHaveBeenCalledWith({
            type: "REMOVE_LABEL_REQUESTED",
            labelId: "UniqueID"
        });
    });

    it("should return updateLabel() in props", () => {
        props.updateLabel("UniqueID", "Label 3", "FooFoo", "#ABCDEF");
        expect(dispatch).toHaveBeenCalledWith({
            type: "UPDATE_LABEL_REQUESTED",
            labelId: "UniqueID",
            name: "Label 3",
            description: "FooFoo",
            color: "#ABCDEF"
        });
    });
});
