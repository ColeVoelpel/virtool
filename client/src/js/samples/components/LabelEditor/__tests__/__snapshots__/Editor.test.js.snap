// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorSelector> should render 1`] = `
<InputGroup>
  <Input__InputLabel>
    Select a color
  </Input__InputLabel>
  <div>
    <ColorSelector__ColorButton
      color="green"
      name="color"
      onClick={[Function]}
    />
    <ColorSelector__ColorButton
      color="orange"
      name="color"
      onClick={[Function]}
    />
    <ColorSelector__ColorButton
      color="yellow"
      name="color"
      onClick={[Function]}
    />
    <ColorSelector__ColorButton
      color="red"
      name="color"
      onClick={[Function]}
    />
    <ColorSelector__ColorButton
      color="purple"
      name="color"
      onClick={[Function]}
    />
    <ColorSelector__ColorButton
      color="blue"
      name="color"
      onClick={[Function]}
    />
    <ColorSelector__ColorButton
      color="primary"
      name="color"
      onClick={[Function]}
    />
  </div>
  <Input__InputLabel>
    or enter a custom color 
  </Input__InputLabel>
  <ColorSelector__ColorInput
    name="color"
    onChange={[Function]}
    placeholder="Ex. #D3D3D3"
    value=""
  />
  <Input__InputError />
</InputGroup>
`;

exports[`<CreateLabel> should render 1`] = `
<div>
  <h3>
    Create a label
  </h3>
  <form>
    <InputGroup>
      <Input__InputLabel>
        Name
      </Input__InputLabel>
      <Input
        error=""
        name="labelName"
        onChange={[Function]}
        placeholder="Label name"
        value=""
      />
      <Input__InputError />
      <Input__InputLabel>
        Description
      </Input__InputLabel>
      <Input
        name="description"
        onChange={[Function]}
        placeholder="Description"
        value=""
      />
    </InputGroup>
    <ColorSelector
      color=""
      errorColor=""
      name="color"
      onColorChange={[Function]}
    />
    <Button
      color="green"
      disabled={false}
      onClick={[Function]}
      tipPlacement="top"
      type="button"
    >
      Create
    </Button>
  </form>
</div>
`;

exports[`<EditLabel> should render 1`] = `
<div>
  <h3>
    Edit a label
  </h3>
  <form>
    <InputGroup>
      <Input__InputLabel>
        Name
      </Input__InputLabel>
      <Input
        error=""
        name="name"
        onChange={[Function]}
        placeholder="Label name"
        value=""
      />
      <Input__InputError />
      <Input__InputLabel>
        Description
      </Input__InputLabel>
      <Input
        name="description"
        onChange={[Function]}
        placeholder="Description"
        value=""
      />
    </InputGroup>
    <ColorSelector
      color=""
      errorColor=""
      name="color"
      onColorChange={[Function]}
    />
    <Button
      color="green"
      disabled={false}
      onClick={[Function]}
      tipPlacement="top"
      type="button"
    >
      Save
    </Button>
    <Button
      color="grey"
      disabled={false}
      onClick={[Function]}
      tipPlacement="top"
      type="button"
    >
      Cancel
    </Button>
  </form>
</div>
`;
